notification/fhir-notification-kafka/src/main/java/com/ibm/fhir/notifications/kafka/impl/FHIRNotificationKafkaPublisher.java

```diff
@@ -79,57 +81,64 @@ private void init(String topicName, Properties kafkaProps) {
-            log.log(Level.SEVERE, msg, t);
+            LOG.log(Level.SEVERE, msg, t);
             throw new IllegalStateException(msg, t);
         } finally {
-            log.exiting(this.getClass().getName(), "init");
+            LOG.exiting(this.getClass().getName(), "init");
         }
     }
 
     /**
      * Performs any necessary "shutdown" logic to disconnect from the topic.
      */
     public void shutdown() {
-        log.entering(this.getClass().getName(), "shutdown");
+        LOG.entering(this.getClass().getName(), "shutdown");
 
         try {
-            if (log.isLoggable(Level.FINE)) {
-                log.fine("Shutting down Kafka publisher for topic: '" + topicName + "'.");
+            if (LOG.isLoggable(Level.FINE)) {
+                LOG.fine("Shutting down Kafka publisher for topic: '" + topicName + "'.");
             }
             if (producer != null) {
                 producer.close();
             }
         } finally {
-            log.exiting(this.getClass().getName(), "shutdown");
+            LOG.exiting(this.getClass().getName(), "shutdown");
         }
     }
 
     @Override
     public void notify(FHIRNotificationEvent event) throws FHIRNotificationException {
-        log.entering(this.getClass().getName(), "notify");
+        LOG.entering(this.getClass().getName(), "notify");
         String topicId = "[" + this.kafkaProps.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG) + "]/" + topicName;
         String jsonString = null;
         try {
             jsonString = FHIRNotificationUtil.toJsonString(event, true);
 
-            if (log.isLoggable(Level.FINE)) {
-                log.fine("Publishing kafka notification event to topic '" + topicId + "',\nmessage: " + jsonString);
+            if (LOG.isLoggable(Level.FINE)) {
+                LOG.fine("Publishing kafka notification event to topic '" + topicId + "',\nmessage: " + jsonString);
             }
 
-            producer.send(new ProducerRecord<String, String>(topicName, jsonString), new KafkaPublisherCallback(event, jsonString, topicId));
-
-            if (log.isLoggable(Level.FINE)) {
-                log.fine("Returned from async kafka send...");
+            boolean aysnc = FHIRConfigHelper.getBooleanProperty(FHIRConfiguration.PROPERTY_KAFKA_SYNC, false);
+            if (aysnc) {
+                producer.send(new ProducerRecord<String, String>(topicName, jsonString), new KafkaPublisherCallback(event, jsonString, topicId));
+                if (LOG.isLoggable(Level.FINE)) {
+                    LOG.fine("Returned from async kafka send...");
+                }
+            } else {
+                RecordMetadata metadata = producer.send(new ProducerRecord<String, String>(topicName, jsonString), new KafkaPublisherCallback(event, jsonString, topicId)).get();
+                if (LOG.isLoggable(Level.FINE)) {
+                    LOG.fine(" Record Produced to Topic '" + metadata.topic() + "' at time " + metadata.timestamp());
```

Comment:
not sure if the leading space was accidental or not

Reply:
Fixed -  not intentional